pkg/model/awsmodel/oidc_provider.go

```diff
@@ -35,43 +43,162 @@ type OIDCProviderBuilder struct {
+		Location:  fi.String("oidc/keys.json"),
+		Name:      fi.String("keys.json"),
+		Base:      fi.String(b.Cluster.Spec.PublicDataStore),
+		Public:    fi.Bool(true),
+	}
+	c.AddTask(keysFile)
+
+	discoveryFile := &fitasks.ManagedFile{
+		Contents:  fi.NewBytesResource(discovery),
+		Lifecycle: b.Lifecycle,
+		Location:  fi.String("oidc/.well-known/openid-configuration"),
+		Name:      fi.String("discovery.json"),
+		Base:      fi.String(b.Cluster.Spec.PublicDataStore),
+		Public:    fi.Bool(true),
 	}
+	c.AddTask(discoveryFile)
 
 	c.AddTask(&awstasks.IAMOIDCProvider{
 		Name:        fi.String(b.ClusterName()),
 		Lifecycle:   b.Lifecycle,
-		URL:         fi.String(issuerURL),
+		URL:         fi.String(serviceAccountIssuer),
 		ClientIDs:   []*string{fi.String(defaultAudience)},
-		Tags:        b.CloudTags(b.ClusterName(), false),
 		Thumbprints: thumbprints,
 	})
 
 	return nil
 }
+
+func buildDicoveryJSON(issuerURL string) ([]byte, error) {
+	d := oidcDiscovery{
+		Issuer:                fmt.Sprintf("%v/", issuerURL),
+		JWKSURI:               fmt.Sprintf("%v/keys.json", issuerURL),
+		AuthorizationEndpoint: "urn:kubernetes:programmatic_authorization",
+		ResponseTypes:         []string{"id_token"},
+		SubjectTypes:          []string{"public"},
+		SigningAlgs:           []string{"RS256"},
+		ClaimsSupported:       []string{"sub", "iss"},
+	}
+	return json.MarshalIndent(d, "", "")
+}
+
+type KeyResponse struct {
+	Keys []jose.JSONWebKey `json:"keys"`
+}
+
+type OIDCKeys struct {
+	SigningKey *fitasks.Keypair
+}
+
+// GetDependencies adds CA to the list of dependencies
+func (o *OIDCKeys) GetDependencies(tasks map[string]fi.Task) []fi.Task {
+	return []fi.Task{
+		o.SigningKey,
+	}
+}
+func (o *OIDCKeys) Open() (io.Reader, error) {
+
+	certBytes, err := fi.ResourceAsBytes(o.SigningKey.Certificate())
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cert: %w", err)
+	}
+	block, _ := pem.Decode(certBytes)
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse cert: %w", err)
+	}
+
+	publicKey := cert.PublicKey
+
+	publicKeyDERBytes, err := x509.MarshalPKIXPublicKey(publicKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to serialize public key to DER format: %v", err)
+	}
+
+	hasher := crypto.SHA256.New()
+	hasher.Write(publicKeyDERBytes)
+	publicKeyDERHash := hasher.Sum(nil)
+
+	keyID := base64.RawURLEncoding.EncodeToString(publicKeyDERHash)
+
+	keys := []jose.JSONWebKey{
+		{
+			Key:       publicKey,
+			KeyID:     keyID,
+			Algorithm: string(jose.RS256),
+			Use:       "sig",
+		},
+	}
+
+	keyResponse := KeyResponse{Keys: keys}
+	jsonBytes, err := json.MarshalIndent(keyResponse, "", "")
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal json: %w", err)
+	}
+
+	return bytes.NewReader(jsonBytes), nil
+}
+
+func getFingerprints() []string {
+
+	return []string{
+		"9e99a48a9960b14926bb7f3b02e22da2b0ab7280",
+		"a9d53002e97e00e043244f3d170d6f4c414104fd",
```

Comment:
Probably worth a comment describing where these came from and how they're derived

Reply:
Made a quick comment about that now.